src/ngAnimate/animateCss.js

```diff
@@ -749,8 +749,8 @@ var $AnimateCssProvider = ['$animateProvider', function($animateProvider) {
           options.onDone();
         }
 
-        // Remove the transitionend / animationend listener(s)
-        if (events) {
+        if (events && events.length) {
```

Comment:
It's actually possible that events is undefined, because if no node is detected, the animation is closed before a bunch of variables are even declared.